src/cmd/run.rs

```diff
@@ -271,6 +309,20 @@ async fn execute_poll_integration<D: Database>(
             // Using `filter_map` instead of `filter` so that this closure gets ownership of `reference`,
             // which makes binding it across an await point easier (no lifetimes to mess with).
             .filter_map(|reference| async {
+                match reference.reference_type() {
+                    git::Reference::Branch {..} => {
+                        // Skipping because integration is not configured to scan branches or branch was not in the integration's watched branches
+                        if !*integration.import_branches() || !integration.validate_reference_scan(reference.name()){
+                            return None
+                        }
+                    },
+                    git::Reference::Tag{..}  => {
+                        // Skipping because integration was not configured to scan tags
+                        if !*integration.import_tags(){
+                            return None
+                        }
```

Comment:
Make sure to run the language formatter on PRs:
```suggestion
                        if !*integration.import_tags() {
                            return None
                        }
```